blockchain/state/statedb.go

```diff
@@ -856,6 +852,7 @@ func (s *StateDB) CreateSmartContractAccountWithKey(addr common.Address, humanRe
 		account.AccountValueKeyCodeInfo:      params.NewCodeInfoWithRules(format, r),
 	}
 	new, prev := s.createObjectWithMap(addr, account.SmartContractAccountType, values)
+	new.created = true
```

Comment:
Just a comment: It'll be applied to precompiled contracts via `CreateSmartContractAccount`. But I think it won't make problem since they're also guaranteed not to be destructed. Introducing a flag to distinguish them seems a little bit overkill.